core/orchestrator.go

```diff
@@ -102,26 +102,43 @@ func (orch *orchestrator) ProcessPayment(payment net.Payment, manifestID Manifes
 		return nil
 	}
 
-	ticket := &pm.Ticket{
-		Recipient:         ethcommon.BytesToAddress(payment.Ticket.Recipient),
-		Sender:            ethcommon.BytesToAddress(payment.Ticket.Sender),
-		FaceValue:         new(big.Int).SetBytes(payment.Ticket.FaceValue),
-		WinProb:           new(big.Int).SetBytes(payment.Ticket.WinProb),
-		SenderNonce:       payment.Ticket.SenderNonce,
-		RecipientRandHash: ethcommon.BytesToHash(payment.Ticket.RecipientRandHash),
+	if payment.TicketParams == nil {
+		return errors.Wrapf(nil, "Payment TicketParams is nil")
 	}
-	seed := new(big.Int).SetBytes(payment.Seed)
 
-	sessionID, won, err := orch.node.Recipient.ReceiveTicket(ticket, payment.Sig, seed)
-	if err != nil {
-		return errors.Wrapf(err, "error receiving ticket for payment %v for manifest %v", payment, manifestID)
+	if payment.Sender == nil {
+		return errors.Wrapf(nil, "Sender address is nil")
 	}
 
-	if won {
-		glog.V(common.DEBUG).Info("Received winning ticket")
-		cachePMSessionID(orch.node, manifestID, sessionID)
+	seed := new(big.Int).SetBytes(payment.TicketParams.Seed)
+
+	ticket := &pm.Ticket{
+		Recipient:         ethcommon.BytesToAddress(payment.TicketParams.Recipient),
+		Sender:            ethcommon.BytesToAddress(payment.Sender),
+		FaceValue:         new(big.Int).SetBytes(payment.TicketParams.FaceValue),
+		WinProb:           new(big.Int).SetBytes(payment.TicketParams.WinProb),
+		SenderNonce:       0,
+		RecipientRandHash: ethcommon.BytesToHash(payment.TicketParams.RecipientRandHash),
 	}
 
+	for _, tsp := range payment.TicketSenderParams {
+
+		ticket.SenderNonce = tsp.SenderNonce
+
+		sessionID, won, err := orch.node.Recipient.ReceiveTicket(
+			ticket,
+			tsp.Sig,
+			seed,
+		)
+		if err != nil {
+			return errors.Wrapf(err, "Error receiving ticket %v for manifest %v", ticket, manifestID)
```

Comment:
If a single ticket is results in an error, the rest of the tickets would be skipped over. I think it would be preferable to log the error and continue with the rest of the tickets in this case. WDYT?

Reply:
Definitely ! Thanks for the thorough review. Learning a lot.